### PR TITLE
Fix for #284: MockFile.Delete does not throw exception when file does not exist.

### PR DESCRIPTION
**Pull request by peter-sereda**
Attempt to fix the subject bug.

### CODE REVIEW

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileDeleteTests.cs [hunk 1]
```diff
@@ -35,5 +36,28 @@ public void MockFile_Delete_ShouldThrowArgumentExceptionIfPathContainsOnlyWhites
...
+        public void MockFile_Delete_ShouldSilentlyReturnIfNonExistingFileInExistingFolder()
+        {
+            var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>()
+            {
+                { XFS.Path("C:\\temp\\exist.txt"), new MockFileData("foobar") },
+            });
+
+            string filePath = XFS.Path("C:\\temp\\somefile.txt");
+
+            // Delete() returns void, so there is nothing to check here beside absense of an exception
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockFileDeleteTests.cs]: Consider using `Assert.DoesNotThrow` to support your comment by an actual assertion :)

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileStreamTests.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ public void MockFileStream_Flush_WritesByteToFile()
         public void MockFileStream_Dispose_ShouldNotResurrectFile()
         {
             var fileSystem = new MockFileSystem();
-            var path = XFS.Path("C:\\test");
+            var path = XFS.Path("C:\\some_folder\\test");
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockFileStreamTests.cs]: Why is this change necessary?

Comment (peter-sereda) [reply System.IO.Abstractions.TestingHelpers.Tests/MockFileStreamTests.cs]: Because of potential bug. Consider this code:
```
            var fileSystem = new MockFileSystem();
            var path = XFS.Path("C:\\test.txt");
            DirectoryInfoBase dir = fileSystem.Directory.GetParent(path);
```
It works ok on Windows but throws NullReferenceException on Linux. And after my changes GetParent() method is called inside File.Delete(), so this test started to fail. Adding a new intermediate folder "some_folder" is the fastest way to make the test work again.

I can create a new issue with this if you think it is a real defect.

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileDeleteTests.cs (hunk 1 above)
Comment (peter-sereda) [reply System.IO.Abstractions.TestingHelpers.Tests/MockFileDeleteTests.cs]: Fixed.

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileStreamTests.cs (hunk 2 above)
Comment (fgreinacher) [reply System.IO.Abstractions.TestingHelpers.Tests/MockFileStreamTests.cs]: Ah OK, thanks. Could you please create an issue for this problem and add a comment to the changed test case that refers to the issue?

Comment (fgreinacher) [reply System.IO.Abstractions.TestingHelpers.Tests/MockFileStreamTests.cs]: Thanks!

**Comment by fgreinacher** [comment]
Thanks again!

**Comment by peter-sereda** [comment]
Welcome!
